server/NetPacksServer.cpp

```diff
@@ -89,7 +89,17 @@ bool SaveGame::applyGh(CGameHandler * gh)
 
 bool EndTurn::applyGh(CGameHandler * gh)
 {
-	PlayerColor player = GS(gh)->currentPlayer;
+	PlayerColor currentPlayer = GS(gh)->currentPlayer;
+	if(player != currentPlayer)
+	{
+		if(gh->getPlayerStatus(player) == EPlayerStatus::INGAME)
+			throwAndComplain(gh, "Player attempted to end turn for another player!");

```

Comment:
will throwAndComplain close the server? Should complain and return false be enough in that case?

Reply:
No, it won't close the server. And this looks to be standard way to abort processing of illegal pack (see rest of the file).

Reply:
It will show green message. And if the player is already lost we treat it as a more or less regular flow and just log warning.
I tried to solve it from AI side but the thing is too asynchronous by design so only with some thread::sleep(100) I was able to correctly shut down AI thread.